src/main/scala/com/whitepages/cloudmanager/state/SolrState.scala

```diff
@@ -128,48 +180,104 @@ case class SolrState(state: ClusterState, collectionInfo: CollectionInfo, config
-    if (nodeList.contains(hostIndicator)) {
-      hostIndicator
+
+  /**
+    *
+    * @param fragment          user passed in string for node identification
+    * @param nodeComparisonMap map where key is the resolved name of a node(IP or Host) and value is its
+    *                          string representation in the cluster state
+    * @return
+    */
+  def unambiguousFragment(fragment: String, nodeComparisonMap: Map[String,String]): Option[String] = {
+    findUnambigousNode(nodeComparisonMap, (s: String) => s == fragment)
+      .orElse(findUnambigousNode(nodeComparisonMap, (s: String) => s.contains(fragment)))
+  }
+
+  /**
+    *
+    * @param nodeComparisonMap map where key is the resolved name of a node(IP or Host) and value is its
+    *                          string representation in the cluster state
+    * @param comparison        function to use for comparison
+    * @return
+    */
+  def findUnambigousNode(nodeComparisonMap: Map[String,String], comparison: (String) => Boolean): Option[String] = {
+    val matchingMaps = nodeComparisonMap.filter{
+      case (resolvedNodeString, clusterNodeString) => comparison(resolvedNodeString) || comparison(clusterNodeString)
     }
-    else {
-      unambiguousFragment(hostIndicator, dnsNameMap(nodeList)).getOrElse {
-        val chunks = hostIndicator.split(':')
-        val host = chunks.head
-        val port = if (chunks.length > 1) ":" + chunks.last else ""
-        val ipAndPort = InetAddress.getByName(host).getHostAddress + port
-        val matches = nodeList.filter((node) => node.contains(ipAndPort))
-        matches.size match {
-          case 0 => throw new ManagerException(s"Could not determine a live node from '$hostIndicator'")
-          case 1 => matches.head
-          case _ => throw new ManagerException(s"Ambiguous node name '$hostIndicator', possible matches: $matches")
+    matchingMaps.toList match {
+      case (resolvedNodeName, clusterNodeString) :: Nil => Some(clusterNodeString)
+      case _ => None
+    }
+  }
+
+
+  /**
+    * This method takes the passed in regular expression and searches for nodes that match this pattern
+    * In the first pass, the pattern attempts to match against fully qualified domain names
+    * If no nodes matched in the first pass, it then attempts to match against the IP addresses
+    * If no nodes matched in the second pass, it then attempts to match against the raw node representation returned by ZK
+    * Return an empty sequence if none of these work
+    *
+    * @param pattern the pattern to use for matching nodes
+    * @param allowOfflineReferences
+    * @return
+    */
+  def getNodeListUsingRegEx(pattern: Regex, allowOfflineReferences: Boolean): Seq[String] = {
+    val clusterStateNodeList = if (allowOfflineReferences) allNodes else liveNodes
+
+    //First try by matching to the fully qualified domain name
+    val dnsNodeList: Map[String, String] = dnsNameMap(clusterStateNodeList)
```

Comment:
Couldn't think of an idiomatic way to implement this without refactoring each individual filter into a method and use case classes